pyvista/core/datasetattributes.py

```diff
@@ -788,7 +788,10 @@ def _prepare_array(
                 raise ValueError(
                     f'data length of ({data.shape[0]}) != required length ({array_len})',
                 )
-
+            if array_len > 0 and data.size == 0:
```

Comment:
This is a better condition since `len` only check the first axis. This change can handle cases like `(0, N)` shape
```suggestion
            if not all(data.shape):
```